Library/src/Multiplayer/SignalRMsgPackEntitySerialiser.cpp

```diff
@@ -27,6 +27,176 @@ namespace csp::multiplayer
 {
 using namespace msgpack_typeids;
 
+namespace
+{
+std::pair<ItemComponentData, signalr::value> ReplicatedValueToSignalRValue(const ReplicatedValue& Value)
+{
+	ItemComponentData ValueType;
+	signalr::value NewValue;
+
+	switch (Value.GetReplicatedValueType())
+	{
+		case ReplicatedValueType::Boolean:
+			ValueType = ItemComponentData::BOOL;
+			NewValue  = signalr::value(Value.GetBool());
+			break;
+		case ReplicatedValueType::Integer:
+			ValueType = ItemComponentData::INT64;
+			NewValue  = signalr::value(Value.GetInt());
+			break;
+		case ReplicatedValueType::Float:
+			ValueType = ItemComponentData::FLOAT;
+			NewValue  = signalr::value(Value.GetFloat());
+			break;
+		case ReplicatedValueType::String:
+			ValueType = ItemComponentData::STRING;
+			NewValue  = signalr::value(Value.GetString().c_str(), Value.GetString().Length());
+			break;
+		case ReplicatedValueType::Vector2:
+		{
+			ValueType	= ItemComponentData::FLOAT_ARRAY;
+			auto VValue = Value.GetVector2();
+			NewValue	= signalr::value(std::vector {signalr::value(VValue.X), signalr::value(VValue.Y)});
+			break;
+		}
+		case ReplicatedValueType::Vector3:
+		{
+			ValueType	= ItemComponentData::FLOAT_ARRAY;
+			auto VValue = Value.GetVector3();
+			NewValue	= signalr::value(std::vector {signalr::value(VValue.X), signalr::value(VValue.Y), signalr::value(VValue.Z)});
+			break;
+		}
+		case ReplicatedValueType::Vector4:
+		{
+			ValueType	= ItemComponentData::FLOAT_ARRAY;
+			auto VValue = Value.GetVector4();
+			NewValue	= signalr::value(
+				   std::vector {signalr::value(VValue.X), signalr::value(VValue.Y), signalr::value(VValue.Z), signalr::value(VValue.W)});
+			break;
+		}
+		case ReplicatedValueType::Map:
+		{
+			ValueType = ItemComponentData::GENERIC_MAP;
+
+			// Creates a custom signalr data structure to hold an array of key-value pairs
+			const auto& Map = Value.GetMap();
+			auto Keys		= Map.Keys();
+
+			// Root of map which contains the type information
+			// and an array of key-value pairs
+			std::vector<signalr::value> MapRoot;
+			MapRoot.reserve(3);
```

Comment:
why 3? (it might be obvious, please forgive me for asking)

Reply:
0 = Type information of the key
1 = Type information of value
2 = vector of key-value pairs

I will update to make this more clear